lec06-pop-models.Rmd

```diff
@@ -2,3 +2,360 @@
+}
+
+radius_earth = 6000 # approximately 6000 km 
+SA_earth = 4*pi*radius_earth^2 # surface area of earth in km^2
+
+surface_covered <- function(N0,r,t) { # calculate the volume taken up by bacteria in m on the earth's surface
+  bac_volume <- N_t(N0,growth_rate,t)*cell_volume # in micrometres cubed
+  height_km = bac_volume*(1e-27)/SA_earth # assume the earth is flat compared to the height of our surface - volume is surface area x height of surface above earth. This is height in km
+  #surface_volume = height*SA_earth
+  
+  height_m = height_km*1000 # 1000 m in km
+  
+  return(height_m)
+}
+
+t<-37
+message("The bacteria would be ", round(surface_covered(2,growth_rate,t),1),  " metres thick on the surface of the earth after ", t, " hours.") #yikes. In less than two days, we would be submerged in bacteria. 
+```
+
+Let's look at another model, the *logistic growth model*, which most of you will have seen before:
+
+$$\frac{dN}{dt} = rN\left(1-\frac{N}{K}\right)$$
+
+$N$ is the number of organisms, $r$ is the growth rate, and $K$ is the carrying capacity of the system. 
+
+Notice that when $N$ is much smaller than $K$, this reduces to the exponential growth model we just looked at since $N/K \approx 0$. 
+
+Let's again plot $dN/dt$ vs. $N$ to find the fixed points:
+
+```{r}
+N_seq <- seq(-20,100) # make a sequence of N values to plot dN/dt against
+
+logistic_eqn <- function(N,r,K) { # calculate dN/dt for the logistic equation
+  return(r*N*(1-N/K))
+}
+
+dN_dt = logistic_eqn(N_seq,r=0.5,K=80)  # notice the vectorized implementation
+
+qplot(N_seq,dN_dt) +
+  geom_line(aes(x=N_seq,y=0*N_seq)) # a line at zero for visual aid
+```
+
+The fixed points are at $N=0$ and $N=80$. If we change $K$, we can figure out that the second fixed point is actually just $N=K$. 
+
+Now we draw the phase portrait (on blackboard):
+
+<!-- Phase portrait for logistic eqn -->
+
+As with the exponential growth model, the fixed point at $0$ is unstable, but now there is also a 
+fixed point at $K$ which is stable. This means that at long times, the system will go towards $N=K$. 
+
+### Note: the importance of stability
+
+Why is stability important? The real world is noisy, and while mathematically an unstable fixed point is 
+still a fixed point (if you were exactly there you would stay there forever), in reality there will 
+always be small perturbations (perhaps a member of the population dies or migrates in) that will move 
+you away from the fixed point. The important thing to know, then, is whether the population will remain 
+close to that fixed point or move away to a different point. 
+
+Think of a ball balanced perfectly on top of a hill, or think of balancing a pen on your finger, 
+or think of balancing a chair on the back two legs. It can stay there as long as there are no disturbances, 
+but the tiniest breath of wind or tremble in your balance will knock it over and it will go towards its other
+fixed point - the ground.
```

Comment:
From you last paragraph, I thought of unstable vs stable fixed points as a ball on top of a hill vs a ball in a (endless?) bowl/valley. Both are stable, but the surrounding landscape either amplifies or dampens the forces of perturbations.

Reply:
The top of the hill isn't stable though. It's unstable because if you move away from the top of the hill, you won't go back towards it.

Reply:
@joelostblom what you're thinking of is that both are in *equilibrium*.

Reply:
Sorry I should have said "Both are *fixed*" instead of "Both are stable". I just meant that if there are no forces acting upon the ball, it will stay on top of the hilll (give that it is exactly at the top of the hill). So that very point is fixed as you said, but wether it is stable or unstable has to do with the properties of the landscape, not the point. This is really just me trying to understand the concept by typing it in different words.

Reply:
What @QuLogic said sounds good, that's probably the term I should have used.

Reply:
@joelostblom yes exactly! Both are *fixed*: if you are exactly on top of it, you will stay there. It's the curvature of the landscape near the fixed point that determines stability, which is why, as we will get to next lecture maybe, you calculate derivatives to assess stability analytically.